core/types/transaction_marshalling.go

```diff
@@ -431,7 +431,63 @@ func (t *Transaction) UnmarshalJSON(input []byte) error {
 				return err
 			}
 		}
-
+	case CeloDenominatedTxType:
```

Comment:
Hmm, shouldn't we marshal the `MaxFeeInFeeCurrency` here as well?

Reply:
Yes, I added all this section for that and forgot to add it